scheduler/pkg/server/utils.go

```diff
@@ -33,3 +35,16 @@ func sendWithTimeout(f func() error, d time.Duration) (bool, error) {
 		return false, err
 	}
 }
+
+func shouldScaleUp(server *store.ServerSnapshot) bool {
+	if server.Stats != nil {
+		stats := server.Stats
+
+		if server.ExpectedReplicas < 0 { // it's not set
```

Comment:
is this possible in practice?

Reply:
I ended up in my PR returning the target replica count to set, which turned out to be easier to test as well. check https://github.com/SeldonIO/seldon-core/blob/99bda5e914c2bd3a70659c01a6ff55e7ed6f17c3/scheduler/pkg/server/utils.go#L48

Reply:
It's possible - there's even a test for this.... 

I'll add the exra return value.